jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -176,27 +173,27 @@ public BaseImageWithAuthorization call()
         } catch (RegistryUnauthorizedException registryUnauthorizedException) {
           // The registry requires us to authenticate using the Docker Token Authentication.
           // See https://docs.docker.com/registry/spec/auth/token
-          RegistryAuthenticator registryAuthenticator =
-              RegistryAuthenticator.initializer(
-                      buildConfiguration.getEventDispatcher(),
-                      buildConfiguration.getBaseImageConfiguration().getImageRegistry(),
-                      buildConfiguration.getBaseImageConfiguration().getImageRepository())
-                  .setAllowInsecureRegistries(buildConfiguration.getAllowInsecureRegistries())
-                  .setUserAgentSuffix(buildConfiguration.getToolName())
-                  .initialize();
-          if (registryAuthenticator == null) {
-            buildConfiguration
-                .getEventDispatcher()
-                .dispatch(
-                    LogEvent.error(
-                        "Failed to retrieve authentication challenge for registry that required token authentication"));
-            throw registryUnauthorizedException;
+          try {
+            RegistryAuthenticator registryAuthenticator =
+                buildConfiguration
+                    .newBaseImageRegistryClientFactory()
+                    .newRegistryClient()
+                    .getRegistryAuthenticator();
+            if (registryAuthenticator != null) {
+              Authorization pullAuthorization =
+                  registryAuthenticator.setCredential(registryCredential).authenticatePull();
```

Comment:
Maybe `authenticatePull()` and `authenticatePush()` should take the credential? There doesn't seem to be any point to setting the credential on the object (it forces us to make it mutable). Wdyt? I don't see us reusing `registryAuthenticator` anywhere anyway.

Reply:
That's a great idea. I actually think `getRegistryAuthenticator()` can accept `Credential` directly.

Reply:
I think there's some code that returns a null `RegistryAuthenticator`, does that matter here?

Reply:
Well, it's simpler to pass `Credential` to `authenticatePush/Pull`. Done.

Reply:
> I think there's some code that returns a null `RegistryAuthenticator`

The code doing the null check on `RegistryAuthenticator`. Are you talking about the `MalformedURLException` case? Seems like no one is throwing this checked exception, so I removed that.

Reply:
Oh, I think I got what you're saying. I decided to pass `Credential` to `authenticatePush/Pull` as you originally suggested.